lib/src/widgets/challenge_list_item.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:dartchess/dartchess.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+import 'package:lichess_mobile/src/model/auth/auth_session.dart';
+import 'package:lichess_mobile/src/model/challenge/challenge.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/common/speed.dart';
+import 'package:lichess_mobile/src/model/lobby/correspondence_challenge.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+import 'package:lichess_mobile/src/styles/lichess_colors.dart';
+import 'package:lichess_mobile/src/styles/lichess_icons.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/user_full_name.dart';
+
+class ChallengeListItem extends ConsumerWidget {
+  const ChallengeListItem({
+    super.key,
+    required this.challenge,
+    required this.user,
+    this.onPressed,
+    this.onCancel,
+  });
+
+  final Challenge challenge;
+  final LightUser user;
+  final VoidCallback? onPressed;
+  final VoidCallback? onCancel;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final me = ref.read(authSessionProvider)?.user;
```

Comment:
Always use `ref.watch` in widgets `build` methods, as recommended in Riverpod's documentation.